src/Analyser/TypeSpecifier.php

```diff
@@ -482,6 +482,8 @@ public function specifyTypesInCondition(
 					$argType = $scope->getType($expr->right->getArgs()[0]->value);
 					if ($argType->isArray()->yes()) {
 						$result = $result->unionWith($this->create($expr->right->getArgs()[0]->value, new NonEmptyArrayType(), $context, false, $scope));
+					} else {
+						$result = $result->unionWith($this->create($expr->right->getArgs()[0]->value, new ConstantArrayType([], []), $context->negate(), false, $scope));
```

Comment:
This fixes `is_string($a) && strlen($a) > 0 || null === $a`. The problem seems to be that in that case `$argType` is still `MixedType` here. I'm not sure why, but this smells a bit like a hacky workaround? If so, I'm fine with removing it and the relevant assertions in order to fix the rest for now. I was not successful debugging everything that is going on leading to `MutatingScope::$resolvedTypes` being set incorrectly..